infrastructure/cloudconnect/gcp-vpn.jinja

```diff
@@ -0,0 +1,126 @@
+{#
+Copyright 2017 Google Inc.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+#}
+
+{% set id = env["deployment"] %}
+{% set region = properties["region"] %}
+{% set bgp = true if properties["asn"] else false %}
+
+resources:
+  - name: {{ id }}
+    type: compute.v1.targetVpnGateway
+    properties:
+      network: {{ properties["network"] }}
+      region: {{ region }}
+ 
+  {% if bgp %}
+  - name: {{ id }}-router
+    type: compute.v1.router
+    properties:
+      network: {{ properties["network"] }}
+      region: {{ region }}
+      bgp:
+        asn: {{ properties["asn"] }}
+  {% endif %}
+
+  - name: {{ id }}-rule-esp
+    type: compute.v1.forwardingRule
+    properties:
+      IPAddress: {{ properties["address"] }}
+      IPProtocol: ESP
+      region: {{ region }}
+      target: $(ref.{{ id }}.selfLink)
+  
+  - name: {{ id }}-rule-udp500
+    type: compute.v1.forwardingRule
+    properties:
+      IPAddress: {{ properties["address"] }}
+      IPProtocol: UDP
+      portRange: 500
+      region: {{ region }}
+      target: $(ref.{{ id }}.selfLink)
+  
+  - name: {{ id }}-rule-udp4500
+    type: compute.v1.forwardingRule
+    properties:
+      IPAddress: {{ properties["address"] }}
+      IPProtocol: UDP
+      portRange: 4500
+      region: {{ region}}
+      target: $(ref.{{ id }}.selfLink)
+ 
+{% for i in properties["tunnels"] %}
+  {% set oindex = loop.index %} 
+  - name:  {{ id }}-tunnel{{oindex}}
+    type: compute.v1.vpnTunnel
+    properties:
+      peerIp: {{ i["peerIp"] }}
+      region: {{ region }}
+      sharedSecret: {{ i["secret"] }}
+      ikeVersion: 1
```

Comment:
interesting that AWS only supports ikev1 https://forums.aws.amazon.com/thread.jspa?messageID=777039